src/main/java/org/json/JSONArray.java

```diff
@@ -1534,12 +1593,25 @@ private void addAll(Collection<?> collection) {
         }
```

Comment:
I just noticed the JSONObject.wrap method here (and in the other `addAll` method), and I think we may have an issue with consistency. If you notice both the `JSONObject.put(String, Object)` and `JSONArray.put(Object)` methods do not do any "wrap" calls to the values. However, the constructors (before the new `addAll` mothods) both wrap values. I feel like this may have been an oversight at some point. Changing the wrap to happen consistently in the `put` methods breaks a number of tests.

@stleary some guidance on what to do in this case would be appreciated.

Reply:
so, to this effect, I actually worked around it in my own copy by adding the wrap calls to the `put` methods